### PR TITLE
[BEAM-551] Add utility to handle JSON option manipulation

### PR DESCRIPTION
**Pull request by sammcveety**
Provide a utility to be used with RuntimeValueProviders that allows runners to manipulate PipelineOptions by interleaving runtime parameters.

Green travis: https://travis-ci.org/sammcveety/incubator-beam/jobs/167702097

R: @dhalperi

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/options/ValueProviderUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  public void testUpdateSerialize() throws Exception {
+    TestOptions submitOptions = PipelineOptionsFactory.as(TestOptions.class);
+    ObjectMapper mapper = new ObjectMapper();
+    String serializedOptions = mapper.writeValueAsString(submitOptions);
+    String updatedOptions = ValueProviderUtils.updateSerializedOptions(
+      serializedOptions, ImmutableMap.of("string", "bar"));
+    TestOptions runtime = mapper.readValue(updatedOptions, PipelineOptions.class)
+      .as(TestOptions.class);
+    assertEquals("bar", runtime.getString());
+  }
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/options/ValueProviderUtilsTest.java]: can you test with a value that gets replaced?

Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/options/ValueProviderUtilsTest.java]: Additionally, with an options with multiple values, one gets preserved and another gets replaced.

And possibly replacing with an empty map and ensuring it works.

Comment (sammcveety) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/options/ValueProviderUtilsTest.java]: Done

**Comment by dhalperi** [comment]
Minor compile error.

```
[ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_MavenVerify@2/sdks/java/core/src/test/java/org/apache/beam/sdk/options/ValueProviderUtilsTest.java:33:10: Redundant 'static' modifier. [RedundantModifier]
Audit done.
[INFO] There is 1 error reported by Checkstyle 6.19 with beam/checkstyle.xml ruleset.
[ERROR] src/test/java/org/apache/beam/sdk/options/ValueProviderUtilsTest.java:[33,10] (modifier) RedundantModifier: Redundant 'static' modifier.
```

Checkstyle has been updated since the commit at which you originally based the PR, perhaps?

**Comment by sammcveety** [comment]
I'd believe that.  Fixed.

**Comment by dhalperi** [comment]
Merged, thanks for fixes.

**Comment by dhalperi** [comment]
(Manually re-ran the jenkins command as this PR was sent when Jenkins was busted)

**Pull request closed without merging**